### PR TITLE
Doc to get around unsupported features

### PR DESCRIPTION
**Pull request by AlejandroLabourdette**
This PR includes:
- Doc to get around unsupported features
- Remove TypeNames expressions from unsupported features

### CODE REVIEW

File: docs/get_around_unsupported_features.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+- Delegate Call / Low Level Call: Cairo Stubs is a way to think about or just replace them with high-level cross-contract calls using interfaces.
+- Dynamic Arrays inside structs: These can be converted to static arrays or perhaps don't use the struct at all. Dynamic Arrays works fine if they are not contained in a struct so perhaps removing the struct is the best approach.
+- Nested tuple expressions: Should be rewrote using solidity to flat it
+- TypeName expressions: No equivalent in Cairo
+- msg.value, msg.sig, msg.data: No equivalent in Cairo so far
+- tx.gasprice, tx.origin: No equivalent in Cairo so far
+- Member access of block(block.gaslimit, block.chainid): No equivalent in Cairo. Most of the time won't be needed.
+- Member access of address (address.balance): To get balance you may have to use the Wrapped ETH token that work as currency for Cairo
+- User defined Errors: No equivalent in Cairo for now.
+- Receive function: Cairo doesn't have a native currency for the protocol so should be rewritten using other approach
```
Comment (swapnilraj) [inline docs/get_around_unsupported_features.md]: Cairo doesn't -> StarkNet doesn't

File: docs/get_around_unsupported_features.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Get around unsupported features
+
```
Comment (swapnilraj) [inline docs/get_around_unsupported_features.md]: Can we split it into two sections, one for the ones that can not be resolved and one for the ones you can get around

File: docs/docs/get_around_unsupported_features.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# Get around unsupported features.
+
+There is no algorithm to get around unsupported features, in most scenarios is needed to understand what is the piece of code doing. There are some unsupported features that are impossible to rewrite right now, understanding what is expected to do when translated to Cairo may indicate that the logic can just be removed.
```
Comment (cicr99) [inline docs/docs/get_around_unsupported_features.md]: Try changing these parts to:
- in most scenarios **it** is necessary to understand what the piece of code **is** doing
- understanding what **it** is expected to do when translated to Cairo

File: docs/docs/get_around_unsupported_features.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+## Unsupported features that can not be resolved.
+
+- Function gasleft(): No way to go around but in most cases won't be needed.
```
Comment (cicr99) [inline docs/docs/get_around_unsupported_features.md]: ... go around **it**, but ...

File: docs/docs/get_around_unsupported_features.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+- Try/Catch: Rewrite the logic using solidity
+- Nested tuple expressions: Should be rewrote using solidity to flat it
```
Comment (cicr99) [inline docs/docs/get_around_unsupported_features.md]: ...should be **rewritten** using...

File: docs/docs/get_around_unsupported_features.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+- tx.gasprice, tx.origin: No equivalent in Cairo so far
+- Member access of block(block.gaslimit, block.chainid): No equivalent in Cairo. Most of the time won't be needed.
+- User defined Errors: No equivalent in Cairo for now.
+- Receive function: StarkNet doesn't have a native currency for the protocol so should be rewritten using other approach
+- Function call options e.g x.f{gas: 10000}(arg1): Most of the options passed are not compatible with Cairo so in most cases can be just removed
+
+## Unsupported features that developer can get around.
+
+- Indexed parameters in events: In this case the index keyword can just be removed
+- Fallback functions with arguments: Must be removed or transformed to a fallback function that receive no parameter
```
Comment (cicr99) [inline docs/docs/get_around_unsupported_features.md]: ... or transformed **into** a ...
... that receive**s** no parameter**s** ...

File: docs/docs/get_around_unsupported_features.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+- Yul blocks: If efficiency is needed then a way to go is Cairo Stubs otherwise may be rewrote with solidity. Sometimes yul blocks are all about arithmetic, this cases are to avoid overflow and underflow errors. Writing that code with solidity enclosed in an unchecked block should be enough.
```
Comment (cicr99) [inline docs/docs/get_around_unsupported_features.md]: - may be **rewritten** with
- arithmetic, **these** cases are

File: docs/docs/get_around_unsupported_features.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+- Delegate Call / Low Level Call: Cairo Stubs is a way to think about or just replace them with high-level cross-contract calls using interfaces.
```
Comment (cicr99) [inline docs/docs/get_around_unsupported_features.md]: you could say:  Cairo Stubs are a ... or Using Cairo Stubs is

File: docs/docs/get_around_unsupported_features.md [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+- Dynamic Arrays inside structs: These can be converted to static arrays or perhaps don't use the struct at all. Dynamic Arrays works fine if they are not contained in a struct so perhaps removing the struct is the best approach.
```
Comment (cicr99) [inline docs/docs/get_around_unsupported_features.md]: Dynamic Arrays **work** fine

File: docs/docs/get_around_unsupported_features.md [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+- Member access of address (address.balance): To get balance you may have to use the Wrapped ETH token that work as currency for Cairo
```
Comment (cicr99) [inline docs/docs/get_around_unsupported_features.md]: that **works** as

File: docs/docs/get_around_unsupported_features.md [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+- Function gasleft(): No way to go around it, but in most cases won't be needed.
+- Try/Catch: Rewrite the logic using solidity
+- Nested tuple expressions: Should be rewritten using solidity to flat it
+- TypeName expressions: No equivalent in Cairo
+- msg.value, msg.sig, msg.data: No equivalent in Cairo so far
+- tx.gasprice, tx.origin: No equivalent in Cairo so far
+- Member access of block(block.gaslimit, block.chainid): No equivalent in Cairo. Most of the time won't be needed.
+- User defined Errors: No equivalent in Cairo for now.
+- Receive function: StarkNet doesn't have a native currency for the protocol so should be rewritten using other approach
+- Function call options e.g x.f{gas: 10000}(arg1): Most of the options passed are not compatible with Cairo so in most cases can be just removed
```
Comment (piwonskp) [inline docs/docs/get_around_unsupported_features.md]: Some bullet points have a dot at the end while others don't. Please put a dot at the end of each bullet point

File: docs/docs/get_around_unsupported_features.md [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+
+## Unsupported features that developer can get around.
+
+- Indexed parameters in events: In this case the index keyword can just be removed
+- Fallback functions with arguments: Must be removed or transformed into a fallback function that receives no parameters
+- Yul blocks: If efficiency is needed then a way to go is Cairo Stubs otherwise may be rewritten with solidity. Sometimes yul blocks are all about arithmetic, these cases are to avoid overflow and underflow errors. Writing that code with solidity enclosed in an unchecked block should be enough.
+- Delegate Call / Low Level Call: Cairo Stubs are a way to think about or just replace them with high-level cross-contract calls using interfaces.
+- Dynamic Arrays inside structs: These can be converted to static arrays or perhaps don't use the struct at all. Dynamic Arrays work fine if they are not contained in a struct so perhaps removing the struct is the best approach.
```
Comment (piwonskp) [inline docs/docs/get_around_unsupported_features.md]: The `perhaps` in the last sentence seems odd. We should give straightforward advices

**Review by piwonskp — APPROVED** [review]

**Review by cicr99 — APPROVED** [review]

**Pull request merged**